### PR TITLE
List item tiles now display a prettier datetime format

### PR DESCRIPTION
**Pull request by chullman**
Formatting is now "_YYYY-MM-DD at HH:mm:ss_" at 24 hour time.

image

Instances of list items whose DateTime is empty (i.e. null) will have a default string of "_0001-01-01 at 00:00:00_" at 24 hour time.

image

### CODE REVIEW

File: lib/common.dart [hunk 1]
```diff
@@ -1,6 +1,22 @@
...
+const DISPLAY_DATE_FORMAT = 'yyyy-MM-dd';
+const DISPLAY_TIME_FORMAT = 'HH:mm:ss'; // 24-hour time format
+
+String displayNeatTimestamp(DateTime? dateTime) {
+  // If DateTime is null, set it to a default of 0001-01-01 at 00:00:00
+  if (dateTime == null) {
+    return Jiffy([0001, 01, 01]).format(DISPLAY_DATE_FORMAT) +
+        " at " +
+        Jiffy([0]).format(DISPLAY_TIME_FORMAT);
+  } else {
```
Comment (kovarny) [inline lib/common.dart:14]: There is no value in this code other than to show an error. And we should not build any logic or write any code to account for some hypothetical app errors in the future.

File: lib/common.dart [hunk 2]
```diff
@@ -1,6 +1,22 @@
+import 'package:cloud_firestore/cloud_firestore.dart';
 import 'package:jiffy/jiffy.dart';
 
-const DATE_FORMAT = 'yyyy-MM-dd';
+// The date at time formats to use for Jiffy DateTime formatting
+const DISPLAY_DATE_FORMAT = 'yyyy-MM-dd';
+const DISPLAY_TIME_FORMAT = 'HH:mm:ss'; // 24-hour time format
+
+String displayNeatTimestamp(DateTime? dateTime) {
```
Comment (kovarny) [inline lib/common.dart:8]: Why making it null and account for null in it in such a way when we can make it just DateTime?

File: lib/common.dart [hunk 3]
```diff
@@ -1,6 +1,22 @@
...
+  // If DateTime is null, set it to a default of 0001-01-01 at 00:00:00
+  if (dateTime == null) {
+    return Jiffy([0001, 01, 01]).format(DISPLAY_DATE_FORMAT) +
+        " at " +
+        Jiffy([0]).format(DISPLAY_TIME_FORMAT);
+  } else {
+    return Jiffy(dateTime).format(DISPLAY_DATE_FORMAT) +
+        " at " +
+        Jiffy(dateTime).format(DISPLAY_TIME_FORMAT);
```
Comment (kovarny) [inline lib/common.dart:17]: Why not have Jiffy(dateTime).format(DISPLAY_DATE_TIME_FORMAT)?

**Review by kovarny — CHANGES_REQUESTED** [review]

**Review by chullman — COMMENTED** [review]
I removed unneeded code to show an error on a hypothetical scenario. Now assumes DateTime will never be null

```dart
String displayNeatTimestamp(DateTime dateTime) {
  // Set DateTime argument to the format of yyyy-MM-dd HH:mm:ss (24-hour time)
  return Jiffy(dateTime).format(DISPLAY_DATE_TIME_FORMAT);
}
```

But in Debug now, entities that don't have the `lastUpdateTime` field will show the red error box:

image

Is this ok?

**Review by kovarny — COMMENTED** [review]
spoke too much

**Review by kovarny — APPROVED** [review]

**Review by kovarny — APPROVED** [review]

**Pull request merged**